src/Plugin/resource/DataProvider/DataProviderInterface.php

```diff
@@ -12,6 +12,18 @@
 use Drupal\restful\Plugin\resource\Field\ResourceFieldInterface;
 
 interface DataProviderInterface extends CrudInterface {
+
+  /**
+   * Return the discovery information for the given entity.
+   *
+   * @param string $path
+   *   The request path.
+   *
+   * @return array
+   *   An array of data for the thing being discovered.
+   */
+  public function discover($path = NULL);
```

Comment:
`getDiscoveryInfo`. `discover` is more direct, but doesn't sound as returning the info. not sure.